src/runtime/providers/math.ts

```diff
@@ -13,7 +13,9 @@ abstract class TotalFunctionConstraint extends Constraint {
   // proposed variable.
   resolveProposal(proposal, prefix) {
     let {args} = this.resolve(prefix);
-    return [this.getReturnValue(args)];
+    let result = this.getReturnValue(args);
+    if (isNaN(result) || !(isFinite(result))) {return [];}
```

Comment:
Removed the check. This was also was what was handling the filtering for Divide by zero, so handled that in Div function.